examples/poll/src/components/Modal/AddQuestionModal.spec.tsx

```diff
@@ -0,0 +1,77 @@
+import { VernaForm, VernaProvider } from '@openfun/verna';
+import * as factories from '@openfun/verna/src/tests/mocks/factories';
+import { act, render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { Suspense } from 'react';
+import AddQuestionModal from ':/components/Modal/AddQuestionModal';
+
+describe('feature to add widgets using a modal and a select', () => {
+  const VernaSuspenseWrapper = ({ idParts }: { idParts: string[] }) => (
+    <Suspense fallback={<span data-testid="suspense-fallback" />}>
+      <VernaProvider
+        isEditor
+        defaultSchema={factories.vernaSchemaFactory()}
+        defaultWidgets={factories.widgetsFactory()}
+        locale="en-US"
+        configSchema={{
+          properties: {
+            selectWidget: {
+              properties: {
+                enum: {
+                  additionalItems: {
+                    type: 'boolean',
+                  },
+                  items: {
+                    type: 'string',
+                  },
+                  type: 'array',
+                },
+              },
+            },
+          },
+        }}
+      >
+        <VernaForm />
+        <AddQuestionModal idParts={idParts} setIsNewPollModalVisible={() => undefined} />
+      </VernaProvider>
+    </Suspense>
+  );
+
+  it('should be able to add a widget from the list with correct type and name', async () => {
+    render(<VernaSuspenseWrapper idParts={['root', 'testSection']} />);
+
+    // Wait that the form is rendered...
+    await waitFor(() => {
+      expect(screen.queryByTestId('suspense-fallback')).toBeNull();
+    });
+
+    // Check that the modal is rendered
+    expect(screen.getByText('Select a type of question to add')).toBeInTheDocument();
+
+    // Add a Multiple choice question field
+    await act(async () => {
+      await userEvent.click(screen.getByRole('button', { name: 'Open Drop' }));
+      await userEvent.click(screen.getByRole('option', { name: 'Single choice question' }));
+      await userEvent.click(screen.getByRole('button', { name: 'Add' }));
+    });
+
+    // Open parameters of the newly created field and add an option
+    await act(async () => {
+      await userEvent.click(screen.getAllByRole('button', { name: 'Parameters' })[3]);
+      await userEvent.click(screen.getAllByRole('button', { name: 'Add' })[0]);
+      await userEvent.type(screen.getAllByRole('textbox', {})[1], 'test');
+    });
+
+    // Save parameters
+    await act(async () => {
+      await userEvent.click(screen.getByRole('button', { name: 'save' }));
+    });
+
+    // Reopen parameters
+    await act(async () => {
+      await userEvent.click(screen.getAllByRole('button', { name: 'Parameters' })[3]);
+    });
+
+    expect(screen.getAllByRole('textbox', {})[1]).toHaveValue('test');
+  });
```

Comment:
I think a better test would be to test if schema has been updated properly once the user clicks to add the widget.

Reply:
Indeed it's more reliable i'll add one